packages/core/tocco-ui/src/Button/Button.spec.js

```diff
@@ -1,97 +1,108 @@
+      const pendingIcon = await waitFor(() => screen.queryByTestId('icon-circle-notch'))
+      expect(pendingIcon).to.not.exist
+    })
 
-        wrapper = enzymeUtil.mountEmbedded(<Button pending />)
-        expect(wrapper.find(LoadingSpinner)).to.have.length(1)
-      })
+    test('should show pending spinner if pending is set', async () => {
+      testingLibrary.renderWithIntl(<Button pending />)
+      const pendingIcon = await waitFor(() => screen.getByTestId('icon-circle-notch'))
+      expect(pendingIcon).to.exist
+    })
 
-      test('should show icon', () => {
-        const wrapper = enzymeUtil.mountEmbedded(<Button icon="icon" />)
-        expect(wrapper.find(Icon)).to.have.length(1)
-      })
+    test('should show icon', async () => {
+      testingLibrary.renderWithIntl(<Button icon="cog" />)
+      const cogIcon = await waitFor(() => screen.getByTestId('icon-cog'))
+      expect(cogIcon).to.exist
+    })
 
-      test('should set default type to button', () => {
-        const wrapper = enzymeUtil.mountEmbedded(<Button />)
-        expect(wrapper.find('button').prop('type')).to.equal('button')
-      })
+    test('should set default type to button', () => {
+      testingLibrary.renderWithIntl(<Button />)
+      expect(screen.getByRole('button').getAttribute('type')).match(/button/)
+    })
 
-      test('should set type', () => {
-        const wrapper = enzymeUtil.mountEmbedded(<Button type="submit" />)
-        expect(wrapper.find('button').prop('type')).to.equal('submit')
-      })
+    test('should set type', () => {
+      testingLibrary.renderWithIntl(<Button type="submit" />)
+      expect(screen.getByRole('button').getAttribute('type')).match(/submit/)
     })
 
     describe('RouterLinkButton', () => {
       test('should show label', () => {
-        const wrapper = enzymeUtil.mountEmbedded(
+        testingLibrary.renderWithIntl(
           <MemoryRouter>
             <RouterLinkButton label="test" />
           </MemoryRouter>
         )
-        expect(wrapper.find('a').text()).to.equal('test')
+        expect(screen.getByText('test')).exist
       })
 
       test('should show children if no label is provided', () => {
         const child = 'test123'
-        const wrapper = enzymeUtil.mountEmbedded(
+
+        testingLibrary.renderWithIntl(
           <MemoryRouter>
             <RouterLinkButton>{child}</RouterLinkButton>
           </MemoryRouter>
         )
-        expect(wrapper.find('a').text()).to.equal(child)
+        expect(screen.getByText(child)).exist
       })
 
       test('should show icon', () => {
-        const wrapper = enzymeUtil.mountEmbedded(
+        testingLibrary.renderWithIntl(
           <MemoryRouter>
-            <RouterLinkButton icon="icon" />
+            <RouterLinkButton icon="icon-cog" />
           </MemoryRouter>
         )
-        expect(wrapper.find(Icon)).to.have.length(1)
+
+        expect(screen.getByRole('link').getAttribute('icon')).match(/icon-cog/)
```

Comment:
suggestion(non-blocking): instead of checking an attribute here we could check the actual icon

```
const cogIcon = await waitFor(() => screen.getByTestId('icon-cog'))
expect(cogIcon).to.exist
```

It's most probably a "bug" that the `icon` is an `attribute` of the `a` Element. It should have been removed since it's a prop not a HtmlAttribute. But there is this issue when we styled `Link` components that sometimes some props will be added to the HtmlElement.

See here:
https://github.com/tocco/tocco-client/commit/80d773d945c7cf89f7a39ec884e2329b80ce9c22